### PR TITLE
Adamantine input file writer

### PR DESCRIPTION
**Pull request by hespuck19**
These changes add an input file writer to mist as well as a unit test function for file creation.

### CODE REVIEW

**Review by stvdwtt — CHANGES_REQUESTED** [review]

File: tests/unit_tests.py [hunk 1]
```diff
@@ -123,7 +123,23 @@ def test_write_pdf(self):
...
+            # Call the function that should create the file
+            mat.write_adamantine_input(adamantine_file_path)
+            
+            # Assert that the file now exists
+            self.assertTrue(os.path.exists(adamantine_file_path), f"File {file} should exist.")
+        
+        except AssertionError as e:
+            self.fail(f"AssertionError occurred: {str(e)}")
+
+def test_write_3dthesis(self):
```
Comment (streeve) [inline tests/unit_tests.py:142]: This test is ignored because of the spacing (I would have expected an error though)

File: mistlib/core.py [hunk 2]
```diff
@@ -303,6 +303,99 @@ def write_pdf(self, file):
 
         return
     
+    def write_adamantine_input(self, file):
+        reference_temperature = self.properties["solidus_eutectic_temperature"].value #this was done for 3dthesis writer prbably can be removed?
+
+        p = self.properties["specific_heat_solid"]
```
Comment (streeve) [inline mistlib/core.py:309]: Looks like some of these properties overlap with 3dthesis - you could create some new `get_property` functions to reduce the repeated code

File: mistlib/core.py [hunk 3]
```diff
@@ -303,6 +303,99 @@ def write_pdf(self, file):
 
         return
     
+    def write_adamantine_input(self, file):
+        reference_temperature = self.properties["solidus_eutectic_temperature"].value #this was done for 3dthesis writer prbably can be removed?
```
Comment (streeve) [inline mistlib/core.py:307]: Probably using the same assumption as below, comment could match

**Review by streeve — CHANGES_REQUESTED** [review]

File: tests/unit_tests.py (hunk 1 above)
Comment (hespuck19) [reply tests/unit_tests.py:142]: Okay I'll fix that.  Thanks!

File: mistlib/core.py [hunk 4]
```diff
@@ -303,6 +303,99 @@ def write_pdf(self, file):
...
+        elif (p.value_type == ValueTypes.LAURENT_POLYNOMIAL):
+                specific_heat_1 = p.evaluate_laurent_polynomial(reference_temperature)
+        else:
+                print("Error")
+
+        p = self.properties["specific_heat_liquid"]
+        if (p.value_type == ValueTypes.SCALAR):
+                specific_heat_2 = p.value
+        elif (p.value_type == ValueTypes.LAURENT_POLYNOMIAL):
+                specific_heat_2 = p.evaluate_laurent_polynomail(reference_temperature)
```
Comment (stvdwtt) [inline mistlib/core.py:321]: Typo:

```suggestion
                specific_heat_2 = p.evaluate_laurent_polynomial(reference_temperature)
```

File: mistlib/core.py (hunk 2 above)
Comment (stvdwtt) [reply mistlib/core.py:309]: That's a good idea -- there should be a function to get the value of a property at a given temperature that hides how it's generated (from a constant value, from a polynomial, from interpolating from a table).

Let's keep that out of scope for this PR, but I'm creating an issue for it.

Comment (stvdwtt) [reply mistlib/core.py:309]: That's now #17.

**Review by stvdwtt — APPROVED** [review]
Everything looks good to me now. @streeve, anything else you'd like before merging?

**Review by streeve — APPROVED** [review]
Looks good

**Pull request merged**